examples/custom-tool-and-agent/README.md

```diff
@@ -0,0 +1,257 @@
+# Custom Tool and Agent Hello World
+This is an example how to create custom tool and agent.
+
+## Pre-requisites
+
+1. **Python v3.11 or above**:
+   - Ensure that Python is installed and available in your environment. You can check using `python --version`.
+     
+      ```bash
+      python --version
+      ```
+
+2. **[Google Cloud CLI](https://cloud.google.com/sdk/docs/install)**:
+   - The `gcloud` CLI is required for authenticating to access private package repositories. Ensure it's installed and configured (`gcloud auth login`, `gcloud config set project <your-project-id>`).
+   - Install the CLI by following the instructions at [cloud.google.com/sdk/docs/install](https://cloud.google.com/sdk/docs/install).
+   - After installation, configure it by running:
+     
+       ```bash
+       gcloud auth login
+       gcloud config set project <your-project-id>
+       ```
+
+3. **VSCode IDE**
+   - Go to [VSCode](https://code.visualstudio.com/download) to download VSCode IDE.
+
+4. **GDP Labs VPN**
+   - The GDP Labs VPN is required to access the GLChat staging environment where tools are uploaded and agents are tested. Ensure you are connected to the VPN before proceeding with steps involving GLChat.
+
+### Installing dependencies
+
+1. Clone the `gen-ai-examples` repository.
+
+   ```bash
+   git clone https://github.com/GDP-ADMIN/gen-ai-examples.git
+   ```
+
+2. Navigate to the example directory:
+
+   ```bash
+   cd gen-ai-examples/examples/custom-tool-and-agent
+   ```
+
+3. Execute the script:
+
+   - For Linux, macOS, or Windows WSL:
+
+     ```bash
+     ./local-start.sh
+     ```
+
+     > [!NOTE]
+     > On Windows, you can either install [WSL (Windows Subsystem for Linux)](https://learn.microsoft.com/en-us/windows/wsl/install) or execute the batch file in Windows Powershell or Command Prompt as described in the next section.
+
+   - For Windows Powershell:
+
+     ```powershell
+     .\local-start.bat
+     ```
+
+   - For Windows Command Prompt:
+
+     ```cmd
+     local-start.bat
+     ```
+
+## Tool Development Guide
+
+### Creating a Custom Tool
+1. Open your VSCode within the `gen-ai-examples/examples/custom-tool-and-agent` directory.
+2. Activate the newly installed virtual environment. VSCode usually activates this environment automatically. If it doesn't activate the environment automatically, then you should click it and select the python from this path `/custom-tool-and-agent/.venv/bin/python`.
+   *   **Alternative**: You can also use the Command Palette (`Ctrl+Shift+P` or `Cmd+Shift+P`), type "Python: Select Interpreter", and choose the interpreter associated with the `.venv` directory within your project.
```

Comment:
For this, I suggest using the same steps like in gen-ai-internal-hello-world.

In local-start.sh (and .bat) there's already log for user to copy paste the python path.